R/write_dwc.R

```diff
@@ -221,22 +231,203 @@ write_dwc <- function(package, directory = ".", doi = package$id,
+    dplyr::mutate(
+      # RECORD LEVEL
+      basisOfRecord = "HumanObservation",
+      dataGeneralizations = NA_character_,
+      # OCCURRENCE
+      occurrenceID = paste(.data$`animal-id`, .data$`tag-id`, "start", sep = "_"), # Same as EventID
+      sex = dplyr::case_when(
+        `animal-sex` == "m" ~ "male",
+        `animal-sex` == "f" ~ "female",
+        `animal-sex` == "u" ~ "unknown"
+      ),
+      lifeStage = .data$`animal-life-stage`,
+      reproductiveCondition = as.logical(.data$`animal-reproductive-condition`),
+      occurrenceStatus = "present",
+      # ORGANISM
+      organismID = .data$`animal-id`,
+      organismName = .data$`animal-nickname`,
+      # EVENT
+      eventID = paste(.data$`animal-id`, .data$`tag-id`, "start", sep = "_"),
+      parentEventID = paste(.data$`animal-id`, .data$`tag-id`, sep = "_"),
+      eventType = "tag attachment",
+      eventDate = format(
+        .data$`deploy-on-date`,
+        format = "%Y-%m-%dT%H:%M:%SZ"
+      ),
+      samplingProtocol = "tag attachment",
+      eventRemarks = paste( # Problems with NA ####
+        dplyr::coalesce(
+          paste2(c(.data$`tag-manufacturer-name`, .data$`tag-model`, "tag")
+          ),
+          paste2(c(.data$`tag-manufacturer-name`, "tag")),
+          "tag"
+        ),
+        dplyr::coalesce(
+          paste2(c("attached by", .data$`attachment-type`, "to")),
+          "attached to"
+        ),
+        dplyr::case_when(
+          `manipulation-type` == "none" ~ "free-ranging animal",
+          `manipulation-type` == "confined" ~ "confined animal",
+          `manipulation-type` == "recolated" ~ "relocated animal",
+          `manipulation-type` == "manipulated other" ~ "manipulated animal",
+          .default = "likely free-ranging animal"
+        ),
+        dplyr::coalesce(
+          paste2(c("|", .data$`deployment-comments`)),
+          "",
+        )
+      ),
+      # LOCATION
+      minimumElevationInMeters = NA_integer_,
+      maximumElevationInMeters = NA_integer_,
+      locationRemarks = NA_character_,
+      decimalLatitude = .data$`deploy-on-latitude`,
+      decimalLongitude = .data$`deploy-on-longitude`,
+      geodeticDatum = dplyr::case_when(
+        !is.null(.data$`deploy-on-latitude`) ~ "EPSG:4326"
+      ),
+      coordinateUncertaintyInMeters = dplyr::case_when(
+        # Assume coordinate precision of 0.001 degree (157m) and recording by GPS (30m)
+        !is.null(.data$`deploy-on-latitude`) ~ 187
+      ),
+      # TAXON
+      scientificName = .data$`animal-taxon`,
+      kingdom = "Animalia",
+      .keep = "none"
+    ) %>%
+    dplyr::left_join(
+      taxa %>%
+        dplyr::mutate(
+          scientificNameID = .data$aphia_lsid,
+          scientificName = .data$name,
+          .keep = "none"
+        ),
+      by = "scientificName"
+    ) %>%
+    dplyr::relocate(
+      .data$scientificNameID,
+      .before = .data$scientificName
+    ) %>%
```

Comment:
I would stop the chain here and save this as `dwc_occurrence_ref`. Then create a `dwc_occurrence_gps`. In the end, do `dplyr::bind_rows()` and mutate with the dataset information.